test/login_test.exs

```diff
@@ -208,16 +208,40 @@ defmodule LoginTest do
   test "server type 'secondary' against two primary instances", context do
     message =
       capture_log(fn ->
-       opts = [
+        opts = [
           endpoints: [{"localhost", 5432}, {"localhost", 5432}],
           target_server_type: :secondary
         ]
 
         assert_start_and_killed(opts ++ context[:options])
       end)
 
-    assert message =~ "** (Postgrex.Error) failed to establish connection to multiple endpoints:\n\n"
-    assert message =~ "  * localhost:5432: (Postgrex.Error) the server type is not as expected. expected: secondary. actual: primary"
+    assert message =~
+             "** (Postgrex.Error) failed to establish connection to multiple endpoints:\n\n"
+
+    assert message =~
+             "  * localhost:5432: (Postgrex.Error) the server type is not as expected. expected: secondary. actual: primary"
+  end
+
+  test "outputs an error message per attempted endpoint when more than one endpoint is used",
+       context do
+    assert capture_log(fn ->
+             opts = [
+               endpoints: [{"doesntexist", 5432}, {"localhost", 5555}],
+               connect_timeout: 100
+             ]
+
+             assert_start_and_killed(opts ++ context[:options])
+           end) =~
+             ~r'\*\* \(Postgrex\.Error\) failed to establish connection to multiple endpoints\:\n\n  \* doesntexist\:5432\: \(DBConnection\.ConnectionError\) tcp connect \(doesntexist\:5432\)\: non\-existing domain \- \:nxdomain\n  \* localhost\:5555\: \(DBConnection\.ConnectionError\) tcp connect \(localhost\:5555\)\: connection refused \- \:econnrefused'
```

Comment:
I don't think those need to be regexes, as using a string will also be a submatch. Can we use strings instead? They should be easier to read. :) Thanks!